safe-global/safe-dao-governance-app
# refactor: reduce RPC requests on polls

## What it solves

Excessive RPC calls on poll

## How this PR fixes it

The `chainId` (from `useChainId`) and `address` (from the new `useAddress`) are passed to each request removing reliance on `signer` requests for them.

The parsing of allocation data has been separated into it's own hook `useVestingData` which is not polled for, whereas the new `useVotingPower` (which relies on the former) is.

## How to test

Open the app as either a Safe App or DApp and observe fewer RPC requests every 15s. Only one allocation fetch should occur per address/per chain.
